instrumentation/kafka-streams/src/test/java/brave/kafka/streams/ITKafkaStreamsTracing.java

```diff
@@ -191,17 +191,17 @@ public void should_create_spans_from_stream_input_and_output_topics() throws Exc
   public void should_create_spans_from_stream_with_tracing_processor() throws Exception {
```

Comment:
would love to have an additional case--probably on the current master--reproducing the current issue, so we are sure it is fixed and we don't step into that again.

Reply:
I tried accessing headers from the integration tests before the fix but did not get the exception maybe due to the way the junit rule sets up the kafka instance? It would be nice to be able to get a test case indeed.

Reply:
I tried to reproduce it as well but I can't manage to make it fail. Could you replicate your failing scenario in a test-case to debug it?  I'm wondering if `new TracingProducessor(delegate)` on our suppliers already protect us from this issue

Reply:
this reproduces it, it's really trivial just point it to a cluster (pre-create the topics) and let it run https://github.com/jorgheymans/kafka-streams-supplier-testcase